src/main/app/src/components/common/OskariKartta.js

```diff
@@ -1,72 +1,34 @@
-import React, { Component } from 'react';
+import React, { useEffect, useCallback } from 'react';
 import { OsoiteParser as op, Localizer as l } from '#/src/tools/Utils';
+import { urls } from 'oph-urls-js';
 import OskariRPC from 'oskari-rpc';
-import { observer, inject } from 'mobx-react';
 
-@inject('urlStore')
-@observer
-class OskariKartta extends Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      channel: undefined,
-    };
-  }
-
-  componentDidMount() {
-    this.initMap();
-  }
-
-  componentWillReceiveProps(nextProps) {
-    this.props = nextProps;
-    this.setLocation();
-  }
-
-  setLocation() {
-    const data = op.getCoreAddress(this.props.osoite);
-    console.log('Triggering map location search for ' + data);
-    if (this.state.channel) {
-      this.state.channel.postRequest('SearchRequest', data);
-    } else {
-      console.log(
-        'Odottamaton virhe. Karttapalvelun channel puuttuu ' +
-          this.props.urlStore.urls.url('kartta.base-url')
-      );
-    }
-  }
-
-  createChannel() {
-    const IFRAME_DOMAIN = this.props.urlStore.urls.url('kartta.base-url');
+const OskariKartta = ({ osoite, postitoimipaikka }) => {
+  function createChannel() {
+    const IFRAME_DOMAIN = urls.url('kartta.base-url');
     const iFrame = document.getElementById('publishedMap');
     const channel = OskariRPC.connect(iFrame, IFRAME_DOMAIN);
-    this.setState({ channel: channel });
     return channel;
   }
 
-  initMap() {
-    const channel = this.createChannel();
-    const _this = this;
+  const initMap = useCallback(() => {
+    const channel = createChannel();
 
     channel.handleEvent('SearchResultEvent', function(data) {
-      if (
-        data.success &&
-        data.result &&
-        data.result.locations &&
-        data.result.locations.length > 0
-      ) {
-        let location = data.result.locations[0];
+      if (data.success && data?.result?.locations?.length > 0) {
+        let _location = data.result.locations[0];
         data.result.locations.forEach((loc) => {
```

Comment:
Pääset kokonaan eroon tuosta _location-muttujasta tähän tapaan.
`
const {lon, lat, name} = data.result.locations.find(loc => postitoimipaikka.toLowerCase() in [loc.region, loc.village]) ?? data.result.locations[0]
`